File: server/src/org/develar/mapsforgeTileServer/ImageFormat.java
package org.develar.mapsforgeTileServer;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public enum ImageFormat {
  WEBP, PNG, VECTOR;

  @NotNull
  String getContentType() {
    switch (this) {
      case WEBP:
        return "image/webp";
      case PNG:
        return "image/png";
      case VECTOR:
        return "application/octet-stream";

      default:
        throw new IllegalStateException();
    }
  }

  @SuppressWarnings("UnusedDeclaration")
  String getFormatName() {
    return this == WEBP ? "webp" : this == PNG ? "png" : "pixi";
  }

  @Nullable
  public static ImageFormat fromName(@Nullable String name) {
    if (name == null || name.isEmpty()) {
      return null;
    }
    else {
      return name.charAt(0) == 'w' ? WEBP : (name.charAt(0) == 'p' ? PNG : VECTOR);
    }
  }
}

File: server/src/ImageFormat.kt
package org.develar.mapsforgeTileServer

import org.develar.mapsforgeTileServer.ImageFormat.WEBP
import org.develar.mapsforgeTileServer.ImageFormat.PNG
import org.develar.mapsforgeTileServer.ImageFormat.VECTOR

public fun imageFormat(name: String?): ImageFormat? {
  if (name == null || name.isEmpty()) {
    return null
  }
  else {
    return if (name.charAt(0) == 'w') WEBP else (if (name.charAt(0) == 'p') PNG else VECTOR)
  }
}

public enum class ImageFormat {
  WEBP
  PNG
  VECTOR

  fun getContentType() = when (this) {
    WEBP -> "image/webp"
    PNG -> "image/png"
    VECTOR -> "application/octet-stream"
    else -> throw IllegalStateException()
  }

  fun getFormatName() = if (this == WEBP) "webp" else if (this == PNG) "png" else "pixi"
}